### PR TITLE
Hannah/python integration

### PR DESCRIPTION
**Pull request by hannah-martinez**
Hey folks!

This merge request allows us to use the Neuroglancer python package! It includes:
* A fix for the typo in bundle file name
* A new BossCredentialsProvider object that allows us to access APL internal datasets by searching for an access token in three ways:
  * Exposes a new function `neuroglancer.set_boss_token(token)` to let you set a token manually
  * If this token is not set, it will search your filesystem for an intern config file and grab the token from there
  * If there is no intern config file, it uses the token 'public'
* Add back the python package tests in the build action. I have commented out the one that does not work. Consensus is that we will ask Google about it.

I would like to squash this merge request when we perform the merge. If someone who has admin access to this repo could double check that squash merging is allowed, that would be great. [Here's how to do that](https://docs.github.com/en/github/administering-a-repository/configuring-commit-squashing-for-pull-requests).

Here's how to test. The instructions [here](https://github.com/jhuapl-boss/neuroglancer/blob/ndmaster/python/README.md) may also be helpful.
* Pull this branch locally
```
git checkout hannah/python-integration
```
* From the root of the repository, install the locally checked out version of `neuroglancer` to `pip`:
```
python setup.py install
```
* Start the python version of the neuroglancer web instance
```
npm run dev-server-python 
```
* Open a new terminal and start a python interpreter instance (or you can do this in a jupyter notebook). Then, run this small script:
```
import neuroglancer
neuroglancer.set_static_content_source(url="http://127.0.0.1:8080")
viewer = neuroglancer.Viewer()
print(viewer)
```
* A URL should be printed. Visit that URL in a browser to view the neuroglancer instance.
* To test that authentication via token is working, try to access an APL internal dataset. I have been using `boss://https://api.bossdb.io/matelsky2020/tutorial/em`. Please try testing all three types of authentication - manual token (`neuroglancer.set_boss_token(token)`), intern config file, and the default public token.

Finally, for other commands you can try in the python interpreter/jupyter notebook, check out [Lindsey's wiki page](https://github.com/aplbrain/wiki/blob/f9421c1021ac9273122295f7878a89adf93091f4/projects/neuroglancer/pythonAPI.md)!

### CODE REVIEW

**Comment by hannah-martinez** [comment]
Sandy checked and squash merging is allowed for this repo. Please squash and merge this PR when it is approved!

**Review by simpksd1 — APPROVED** [review]

File: python/neuroglancer/boss_credentials.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                    return self._credentials
+
+                # If not, look for config file in intern file location
+                config_path = '~/.intern/intern.cfg'
+                if os.path.isfile(os.path.expanduser(config_path)):
+                    with open(os.path.expanduser(config_path), 'r') as config_file_handle:
+                        config_parser = ConfigParser()
+                        config_parser.readfp(config_file_handle)
+                        try:
+                            self._credentials = config_parser["Default"]["token"]
```
Comment (kitchell) [inline python/neuroglancer/boss_credentials.py:38]: we should add the capability to look in ['Volume Service'] if no default

File: python/neuroglancer/boss_credentials.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        # Make sure logging is initialized.  Does nothing if logging has already
+        # been initialized.
+        logging.basicConfig()
+
+        self._lock = threading.Lock()
+        self._credentials = None
+
+    def set_token(self, token):
+        self._credentials = dict(tokenType=u'Token', accessToken=token)
```
Comment (kitchell) [inline python/neuroglancer/boss_credentials.py:21]: Might be good to add a comment, or type hinting that token should be a string

File: python/neuroglancer/default_credentials_manager.py [hunk 3]
```diff
@@ -26,3 +27,11 @@
...
     u'gcs',
     lambda _parameters: google_credentials.get_google_application_default_credentials_provider())
+
+default_credentials_manager.register(
+    u'boss',
+    lambda _parameters: boss_credentials_provider
+)
+
+def set_boss_token(token):
+    boss_credentials_provider.set_token(token)
```
Comment (kitchell) [inline python/neuroglancer/default_credentials_manager.py:37]: Might be good to add a note, or type hinting that token should be a string

**Review by kitchell — CHANGES_REQUESTED** [review]

File: python/neuroglancer/boss_credentials.py (hunk 1 above)
Comment (hannah-martinez) [reply python/neuroglancer/boss_credentials.py:38]: Fixed, thanks for finding the issue

File: python/neuroglancer/boss_credentials.py (hunk 2 above)
Comment (hannah-martinez) [reply python/neuroglancer/boss_credentials.py:21]: Added!

File: python/neuroglancer/boss_credentials.py (hunk 1 above)
Comment (kitchell) [reply python/neuroglancer/boss_credentials.py:38]: should it be in the except for the first try/except?

Comment (kitchell) [reply python/neuroglancer/boss_credentials.py:38]: no, never mind, I missed the return in the first try, I think this is fine

**Review by kitchell — APPROVED** [review]

**Pull request merged**